### ISSUE TITLE
undefined reference problem in compiling the COAWST case JOE_TC

### ISSUE THREAD

**Issue by chunlei666** [jcwarner-usgs/COAWST#158]
I compiled the SWAN+RMOS successfully, however, errors occured when complied SWAN+ROMS+WRF in the final step.  it gave me module_io_quilt.f90:(.text+0xaffb): undefined reference to `ext_pnc_inquire_filename_'
module_io_quilt.f90:(.text+0xb023): undefined reference to `ext_pnc_ioclose_'
i attached the log and the build_coawst.sh 
 i'll appreciate  for your time to help me solve the problem 
[build_coawst.txt](https://github.com/DOI-USGS/COAWST/files/12354587/build_coawst.txt)
[log.txt](https://github.com/DOI-USGS/COAWST/files/12354588/log.txt)

**Comment by chunlei666** [comment]
COAWST version 3.8

**Comment by chunlei666** [comment]
COAWST version 3.8

**Comment by chunlei666** [comment]
I modified the SWAN/switch.pl and the CmakeLists.txt, because my SWAN compiliation just stoped at the beginning and my friend told me there are some problems in perl. But i dont think this is the reason, because it seems some errors in WRF coupling with ROMS

**Comment by chunlei666** [comment]
Solved, do not use the pnetcdf to compile only by the netcdf.  Seems the WRF in COAWST do not support the parallel IO.

**Comment by ayomidetobi** [comment]
I don understand how you solved this am getting almost same error

**Comment by jcwarner-usgs** [comment]
i think there might have been 2 issues in this post.
What is the error you are getting?  is it with pnetcdf?
you need to set an env var , try
export PNETCDF=""
tell me if that works

**Comment by ayomidetobi** [comment]
[build_coawst.log](https://github.com/user-attachments/files/18463782/build_coawst.log)
here is my log please me look into it  around line 4500

**Comment by ayomidetobi** [comment]
I tried this "export PNETCDF=""
tell me if that works" it didn't work

**Comment by jcwarner-usgs** [comment]
i think you are having a different error.
Can you try to compile the test case with WRF only?
just use
#define WRF_MODEL

**Comment by ayomidetobi** [comment]
The problem is compiling wrf am getting the same error in my log file on line 4500 to 4600
Around

**Comment by ayomidetobi** [comment]
the problem is compiling the wrf that what is causing the error I did everything according to the documentation
this is the error am getting
mpif90  -o wrf.o -c -O2 -ftree-vectorize -funroll-loops -w -ffree-form -ffree-line-length-none -fconvert=big-endian -frecord-marker=4 -fallow-argument-mismatch -fallow-invalid-boz   -I../dyn_em  -I/opt/COAWST/WRF/external/esmf_time_f90  -I/opt/COAWST/WRF/main -I/opt/COAWST/WRF/external/io_netcdf -I/opt/COAWST/WRF/external/io_int -I/opt/COAWST/WRF/frame -I/opt/COAWST/WRF/share -I/opt/COAWST/WRF/phys -I/opt/COAWST/WRF/wrftladj -I/opt/COAWST/WRF/chem -I/opt/COAWST/WRF/inc -I/usr/local/netcdf/include -I/opt/COAWST/Build_roms  -I/opt/COAWST/Lib/MCT/include   wrf.f90
ranlib  libwrflib.a
mpif90  -o wrf.exe  -O2 -ftree-vectorize -funroll-loops -w -ffree-form -ffree-line-length-none -fconvert=big-endian -frecord-marker=4 -fallow-argument-mismatch -fallow-invalid-boz    wrf.o ../main/module_wrf_top.o libwrflib.a  /opt/COAWST/WRF/external/fftpack/fftpack5/libfftpack.a /opt/COAWST/WRF/external/io_grib1/libio_grib1.a /opt/COAWST/WRF/external/io_grib_share/libio_grib_share.a /opt/COAWST/WRF/external/io_int/libwrfio_int.a -L/opt/COAWST/WRF/external/esmf_time_f90 -lesmf_time /opt/COAWST/WRF/external/RSL_LITE/librsl_lite.a /opt/COAWST/WRF/frame/module_internal_header_util.o /opt/COAWST/WRF/frame/pack_utils.o -L/opt/COAWST/WRF/external/io_netcdf -lwrfio_nf -L/usr/local/netcdf/lib -lnetcdff -lnetcdf      -L/usr/local/hdf5-hdf5_1.14.5/lib -lhdf5hl_fortran -lhdf5_hl -lhdf5_fortran -lhdf5 -lm -lz     -L/opt/COAWST/Lib/MCT/lib -lmct -lmpeu -L/usr/local/netcdf/lib -lnetcdf 
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/Scrt1.o: in function `_start':
(.text+0x1b): undefined reference to `main'
/usr/bin/ld: /tmp/ccX4JVPw.ltrans15.ltrans.o: in function `__module_wps_io_arw_MOD_read_wps':
<artificial>:(.text+0xef83fb): undefined reference to `__module_optional_input_MOD_flag_st000010'
/usr/bin/ld: <artificial>:(.text+0xef8408): undefined reference to `__module_optional_input_MOD_flag_st010040'
/usr/bin/ld: <artificial>:(.text+0xef8415): undefined reference to `__module_optional_input_MOD_flag_st040100'
/usr/bin/ld: <artificial>:(.text+0xef8422): undefined reference to `__module_optional_input_MOD_flag_st100200'
/usr/bin/ld: <artificial>:(.text+0xef842f): undefined reference to `__module_optional_input_MOD_flag_sm000010'
/usr/bin/ld: <a

**Comment by jcwarner-usgs** [comment]
for COAWST, we do not need the wrfmain.  so that is not a problem.

**Comment by jcwarner-usgs** [comment]
the other problem is :
/usr/bin/ld: /tmp/ccX4JVPw.ltrans15.ltrans.o: in function __module_wps_io_arw_MOD_read_wps': <artificial>:(.text+0xef83fb): undefined reference to __module_optional_input_MOD_flag_st000010'
etc.
i have not seen this error before.
it might have to deal with wrf having some different compiler flags.

**Comment by jcwarner-usgs** [comment]
can i see a full build output for the option of just
#define WR_MODEL

**Comment by ayomidetobi** [comment]
[build_coawst.log](https://github.com/user-attachments/files/18470966/build_coawst.log)
 here is the log for just wrf 
all other executables are build except wrf.exe
what could be wrong.

by the way  I fix the first issue by recompiling mpich  with same compiler

**Comment by jcwarner-usgs** [comment]
did you get a coawstM ?

**Comment by ayomidetobi** [comment]
Yes I did , am just wondering why the wrf.exe was not compiled ?

**Comment by jcwarner-usgs** [comment]
ok so i just downloaded a new version of the code.
i change JOE_TC/Coupled/joe_tc.h top from:

# undef ExpA1
# undef ExpA
# undef ExpB
# undef ExpC
# undef ExpD
# undef ExpE
# undef ExpF
# define ExpG
#ifdef ExpA1            /* WRF<->SWAN */
# define WRF_MODEL
# define SWAN_MODEL
#endif

to

# define ExpA1
# undef ExpA
# undef ExpB
# undef ExpC
# undef ExpD
# undef ExpE
# undef ExpF
# undef ExpG

#ifdef ExpA1            /* WRF<->SWAN */
# define WRF_MODEL
# undef  SWAN_MODEL
#endif

so this is a WRF only compile.
I selected 15 for ifort and then 1 (no move nest).

It compiled. I got a wrf.exe and a coawsM.
if i do ls-ltr
....
-rwxrwxr-x 1 jwarner sg-usgs-share   15763 Jan 21 11:03 build_coawst.sh
drwxrwsr-x 2 jwarner sg-usgs-share    4096 Jan 21 11:49 Build_roms
drwxrwsr-x 2 jwarner sg-usgs-share    4096 Jan 21 11:50 WRF
lrwxrwxrwx 1 jwarner sg-usgs-share      16 Jan 21 12:24 coawstM -> WRF/main/wrf.exe

if you type
ls -ltr
do you see that ?  did you get a new coawstM?   
did it soft link to WRF/main/wrf.exe?

i dont think it did for you, because you did not get a wrf.exe.
what did your joe_tc.h look like?
i cant see why you did not get a wrf.exe.
